src/averageAges.js

```diff
@@ -14,12 +14,22 @@
  *
  * @return {number}
  */
+
 function calculateMenAverageAge(people, century) {
-  // write code here
-  // learn how to use array methods like .filter .map .some .every .find .reduce
-  // avoid using loop and forEach
-  // replace `if ()` statement with &&, || or ?:
-  // without nesting
+  const men = people.filter(person =>
+    (person.sex === 'm') && (
+      century ? Math.ceil(person.died / 100) === century : true));
+
+  const ages = men.map((person) => calculateAge(person.born, person.died));
+
+  const totalAge = ages.reduce((sum, age) => sum + age, 0);
+  const averageAgeMen = totalAge / ages.length;
+
+  return averageAgeMen;
```

Comment:
All this 4 lines are duplicating
```
  const getPeopleAverageAge = (people) => {
    const ages = people.map((person) => person.died - person.born));

    const totalAge = ages.reduce((sum, age) => sum + age, 0);

    return totalAge / ages.length;
  }
```

```suggestion
  return getPeopleAverageAge(men);
```